### PR TITLE
Release v3.0.0

### PR DESCRIPTION
**Pull request by fsimonis**
## How to work with this template

* [x] assign a release manager, who takes care of the process
* [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.

## This is a maturing draft!

This is a draft for the release and is meant as a target for development and testing of downstream components. We gather status and issues in the comments as well as in the [GitHub project](https://github.com/orgs/precice/projects/13).

This release branch may be rebased occasionally.

## Still to do

* [x] Merge #1899 into develop
* [x] Merge #1675 into develop
* [x] Rebase and prepare changelogs (also include hotfix changelog)
* [ ] Do testing and porting
* [x] Release hotfix 2.5.1

## Step by step guide

* [x] Look over [PRs](https://github.com/precice/precice/pulls?q=is%3Apr+no%3Amilestone+is%3Aclosed) and [Issues](https://github.com/precice/precice/issues?q=is%3Aissue+no%3Amilestone+is%3Aclosed) without an assigned version. (all)
* [x] Look over entries in [`docs/changelog`](https://github.com/precice/precice/blob/develop/docs/changelog)) (all)
   * Add missing entries, if necessary
   * Fix wording and tense
* [x] Make sure you have the latest `develop` and `main` branches locally.
* [x] Merge `main` to `develop` ( This should result in no commits )
* [x] Check code base w.r.t code formatting (run [`precice/tools/formatting/check-format`](https://github.com/precice/precice/blob/develop/tools/formatting/check-format)) and reformat if required (run [`precice/tools/formatting/format-all`](https://github.com/precice/precice/blob/develop/tools/formatting/format-all))
* [x] Update the list of operating systems for the package generation in `.github/workflows/release.yml`
* [x] Create branch `release-vX.Y.Z` from develop. If needed, `git rebase develop`.
* [x] Run `tools/releasing/bumpversion.sh MAJOR.MINOR.PATCH` to bump the version
* [x] Look over the updated `CHANGELOG.md` of the release branch (all)
   * Check for merged lines
   * Add things, if necessary
   * Fix wording and tense
   * Sort the entries lexicographically
   * Extract summary
* [x] Verify the version changes in:
   * [x] `CHANGELOG`
   * [x] `CMakeLists.txt`
   * [x] `tools/releasing/packaging/debian/changelog`
* [x] Commit the version bump
* [x] Push the release branch to the precice repository
* Prepare independent releases @IshaanDesai
   * [x] [Python bindings](https://github.com/precice/python-bindings/blob/develop/docs/ReleaseGuide.md) 
   * [x] [MATLAB bindings](https://github.com/precice/matlab-bindings/blob/develop/docs/ReleaseGuide.md)
   * [x] [JULIA bindings](https://github.com/precice/PreCICE.jl)
* [x] Open PR from `release-vX.Y.Z` to `main` (use [this template](https://github.com/precice/precice/blob/develop/tools/releasing/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
* [x] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [x] Fix potential problems in develop (all)
* [x] Rebase the release branch on develop (all)
* [x] Commit the version bump
* [ ] Draft message to mailing list
* [ ] Write a draft "blog post" on [Discourse](https://precice.discourse.group/)
* [x] Update documentation (all)
   * [x] Update [XML configuration reference](https://github.com/precice/precice.github.io/blob/master/_includes/xmlreference.md) https://github.com/precice/precice.github.io/pull/328
   * [x] Update version in [precice/precice.github.io](https://github.com/precice/precice.github.io):
      * `_config.yml`
      * `_data/sidebars/docs_sidebar.yml`
* [x] Approve the PR with at least two reviews (all)
* [ ] Merge PR to `main` ( use `git merge --no-ff release-vX.Y.Z` )
* [ ] Tag release on `main` `vX.Y.Z` and verify by running `git describe --tags`
* [ ] Merge back to `develop` and verify by running `git describe --tags`
* [ ] Triple check that you haven't messed anything up. (You can always discard local changes)
* [ ] Push `main` and push the `vX.Y.Z` tag
* [ ] Push `develop`
* [ ] Wait for the release pipeline
  * [ ] [To create a new draft release on GitHub](https://github.com/precice/precice/releases)
  * [ ] To automatically generate packages for the latest Debian and the two latest Ubuntu LTS versions.
* [ ] Write release text
* [ ] Publish the GitHub release

## Regression Tests

Use the following branches:
* precice `release-vX.Y.Z`
* pyprecice `python-bindings-vX.Y.Z.1`
* matlab-bindings `matlab-bindings-vX.Y.Z.1`
* rest `main`(`master`)

Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289 and update the table.

| State | Success | Failure | Skipped |
| --- | --- | --- | --- |
| Write | `:o:` | `:x:` | `:fast_forward:` |
| Read | :o: | :x: | :fast_forward: |

| State | Tester | Test |
| --- | --- | --- |
| :o: | @davidscn  | [quickstart](https://github.com/precice/tutorials/tree/master/quickstart) fluid-openfoam - solid-cpp |
| :o: | @BenjaminRodenberg  | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-openfoam - solid-fenics (serial + parallel) |
| :fast_forward: | | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-openfoam - solid-dune |
| :o: | @MakisH | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-nutils - solid-calculix |
| :o: | @davidscn  | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-openfoam - solid-dealii |
| :fast_forward: | su2  | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) ~~fluid-su2 - solid-fenics~~ |
| :o: | @davidscn | :exclamation: [multiple-perpendicular-flaps](https://github.com/precice/tutorials/tree/master/multiple-perpendicular-flaps) fluid-openfoam - solid-(left+right)-dealii |
| :o:/:o: | @davidscn  | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-openfoam serial + parallel |
| :o: | @BenjaminRodenberg | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-fenics serial + parallel |
| :o: | @BenjaminRodenberg  | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-nutils |
| :o: | @davidscn  | [flow-over-heated-plate-nearest-projection](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate-nearest-projection) fluid-openfoam - solid-openfoam |
| :fast_forward: | | [flow-over-heated-plate-steady-state](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate-steady-state) fluid-openfoam - solid-codeaster |
| :o: | @davidscn | :exclamation: [flow-over-heated-plate-partitioned-flow](https://github.com/precice/tutorials/tree/develop/flow-over-heated-plate-partitioned-flow) openfoam-openfoam |
| :fast_forward:  | | [flow-over-heated-plate-two-meshes](https://github.com/precice/tutorials/tree/develop/flow-over-heated-plate-two-meshes) fluid-openfoam - solid-calculix |
| :o:/:o: | @MakisH  | :exclamation: [heat-exchanger](https://github.com/precice/tutorials/tree/master/heat-exchanger) fluid-(inner+outer)-openfoam - solid-calculix |
| :o:/:o:[^1] | @MakisH  | :exclamation: [heat-exchanger-simplified](https://github.com/precice/tutorials/tree/master/heat-exchanger) fluid-(inner+outer)-openfoam - solid-calculix |
| :o: | @fsimonis  | [partitioned-elastic-beam](https://github.com/precice/tutorials/tree/master/partitioned-elastic-beam) dirichlet-calculix - neumann-calculix |
| :o: | @BenjaminRodenberg  | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) fenics - nutils |
| :o: | @BenjaminRodenberg  | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) nutils - nutils |
| :o: | @BenjaminRodenberg  | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) fenics - fenics |
| :o: | @BenjaminRodenberg  | [partitioned-heat-conduction-complex](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction-complex) fenics- fenics |
| :o: | @MakisH | [partitioned-pipe](https://github.com/precice/tutorials/tree/master/partitioned-pipe) fluid1-openfoam-pimplefoam - fluid2-openfoam-sonicliquidfoam |
| :o: | @fsimonis | [elastic-tube-1d](https://github.com/precice/tutorials/tree/master/elastic-tube-1d) fluid-cpp - solid-python |
| :o: | @IshaanDesai  | [elastic-tube-3d](https://github.com/precice/tutorials/tree/master/elastic-tube-3d) fluid-openfoam - solid-calculix |
| :o: | @IshaanDesai  | MATLAB / MATLAB [ODEs](https://github.com/precice/matlab-bindings/tree/develop/tutorial) |
| :fast_forward: | @MakisH | Solverdummy [Fortran module](https://github.com/precice/fortran-module/tree/develop/examples/solverdummy) |
| :o: | @BenjaminRodenberg  | Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy) |
| :o: | @IshaanDesai  | Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy) |
| :o: | @IshaanDesai  | Solverdummy [Julia](https://github.com/precice/PreCICE.jl/tree/develop/solverdummy) |
| :o: | @uekerman  | Alya |
| :o: | @davidscn | SuperMUC (ASTE) |
| :o: | @MakisH |  [channel-transport](https://github.com/precice/tutorials/tree/develop/channel-transport) fluid-openfoam - transport nutils |
| :fast_forward: | adapter todos | [channel-transport-reaction](https://github.com/precice/tutorials/tree/develop/channel-transport-reaction) ~~chemical-fenics - fluid-fenics~~ |
| :o: | @davidscn | [turek-hron-fsi3](https://github.com/precice/tutorials/tree/develop/turek-hron-fsi3) fluid-openfoam - solid-dealii |
| :o: | @IshaanDesai | [two-scale-heat-conduction](https://github.com/precice/tutorials/tree/develop/two-scale-heat-conduction) macro-nutils - micro-nutils |
| :fast_forward:  |  | [volume-coupled-flow](https://github.com/precice/tutorials/tree/develop/volume-coupled-flow) fluid-openfoam - source-nutils |

[^1]: slightly different results in parallel

## Post-release

* [ ] Update version specific documentation
* [ ] Flag [Arch Linux AUR package](https://aur.archlinux.org/packages/precice) and dependants as out-of-date.
* [ ] Update Spack recipe @fsimonis 
* [ ] Update `pyprecice` Spack
* [ ] Update Website:
    * [ ] Bump version in [`_config.yml`](https://github.com/precice/precice.github.io/blob/master/_config.yml)
    * [ ] Update the [XML reference](https://github.com/precice/precice.github.io/blob/master/_includes/xmlreference.md) using `binprecice md`
    * [ ] Look over the [Roadmap](https://www.precice.org/fundamentals-roadmap.html) and update entries.

### Release new version for bindings (to ensure compatibility with newest preCICE version)

- [ ] [Fortran module](https://github.com/precice/fortran-module/compare/master...develop) @MakisH 
- [ ] [MATLAB bindings](https://github.com/precice/matlab-bindings/blob/develop/docs/ReleaseGuide.md) @IshaanDesai (ready)
- [ ] [python bindings](https://github.com/precice/python-bindings/blob/develop/docs/ReleaseGuide.md) @IshaanDesai (ready)
- [ ] [Julia bindings](https://github.com/precice/PreCICE.jl) @IshaanDesai (ready)

### (only if breaking changes) Open PRs or issues `develop -> main` for all adapters

- [ ] [athlet-adapter](https://github.com/precice/athlet-adapter/compare/master...develop)
- [ ] [calculix-adapter](https://github.com/precice/calculix-adapter/compare/master...develop)
- [ ] [code_aster-adapter](https://github.com/precice/code_aster-adapter/compare/master...develop)
- [ ] [comsol-adapter](https://github.com/precice/comsol-adapter/compare/master...develop)
- [ ] [dealii-adapter](https://github.com/precice/dealii-adapter/compare/master...develop)
- [ ] [fenics-adapter](https://github.com/precice/fenics-adapter/compare/master...develop)
- [ ] [fluent-adapter](https://github.com/precice/fluent-adapter/compare/master...develop)
- [ ] lsdyna-adapter?
- [ ] mbdyn-adapter?
- [ ] [openfoam-adapter](https://github.com/precice/openfoam-adapter/compare/master...develop)
- [ ] [su2-adapter](https://github.com/precice/su2-adapter/compare/master...develop)

### (only if breaking changes) Open PRs or issues `develop -> main` for all other tools

- [ ] [aste](https://github.com/precice/aste/compare/master...develop) @davidscn (ready)
- [ ] [fmi-runner](https://github.com/precice/aste/compare/master...develop) @uekerman 
- [ ] [tutorials](https://github.com/precice/tutorials/compare/master...develop)

### Marketing

* [ ] Finalize post on [Discourse](https://precice.discourse.group/)
* [ ] Write on [Matrix](https://matrix.to/#/#precice_lobby:gitter.im?web-instance[element.io]=app.gitter.im)
* [ ] Send announcement to the [mailing list](https://mailman.informatik.uni-stuttgart.de/mailman/listinfo/precice)
* [ ] CFD Online:
     * [ ] [News](https://www.cfd-online.com/Forum/news.cgi/form/0)
     * [ ] Forum (choose a central topic)
* [ ] NADigest
* [ ] [NAFEMS](https://www.nafems.org/mynafems/submitnews/) (needs account, appears in [Upcoming Industry Events](https://www.nafems.org/events/industry-events/))
* [ ] Post on [Twitter](https://twitter.com/preCICE_org) (additionally to the automatic)
* [ ] Post on [Mastodon](https://fosstodon.org/@precice)
* [ ] Post on [ResearchGate](https://www.researchgate.net/project/preCICE)
* [ ] Post in LinkedIn. Relevant places:
    * [ ] [preCICE group](https://www.linkedin.com/groups/9073912/)
    * [ ] [OpenFOAM group](https://www.linkedin.com/groups/1920608/)
    * [ ] [HPC group](https://www.linkedin.com/groups/87791/)
    * [ ] Personal
* [ ] reddit [r/cfd](https://www.reddit.com/r/CFD/)
* [ ] Submit a short article to the [Quartl](https://www.in.tum.de/en/i05/further-activities/quartl/)

### Misc

* [ ] Update the [PR template](https://github.com/precice/precice/blob/develop/tools/releasing/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)

### CODE REVIEW

**Comment by fsimonis** [comment]
The remaining features have been merged and I also updated the thirdparty fmt to the recent 10.2.1.

The changelog is now merged too. Version 2.5.1 is still missing, I will add this later as the intention remains to release 2.5.1 before 3.0.0.

The release is now ready for testing.

@IshaanDesai volunteered to update the test overview.

**Comment by IshaanDesai** [comment]
Upon some thought, running all the tests in the PR description except Alya and SuperMUC (due to lack of access or non-compatibility with v3.0.0) would substantially cover existing solver combinations and specialized variants of tutorial cases. Additionally, I propose manually running the following tests:

| State | Tester | Test |
| -- | -- | -- |
|  |  | [channel-transport](https://github.com/precice/tutorials/tree/develop/channel-transport) fluid-openfoam - transport nutils |
|  |  | [channel-transport-reaction](https://github.com/precice/tutorials/tree/develop/channel-transport-reaction) chemical-fenics - fluid-fenics |
|  |  | [flow-over-heated-plate-partitioned-flow](https://github.com/precice/tutorials/tree/develop/flow-over-heated-plate-partitioned-flow) |
|  |  | [flow-over-heated-plate-two-meshes](https://github.com/precice/tutorials/tree/develop/flow-over-heated-plate-two-meshes) |
|  |  | [mutliple-perpendicular-flaps](https://github.com/precice/tutorials/tree/develop/multiple-perpendicular-flaps) |
|  |  | [turek-hron-fsi3](https://github.com/precice/tutorials/tree/develop/turek-hron-fsi3) |
|  |  | [two-scale-heat-conduction](https://github.com/precice/tutorials/tree/develop/two-scale-heat-conduction) |
|  |  | [volume-coupled-flow](https://github.com/precice/tutorials/tree/develop/volume-coupled-flow) |

Some tutorials are proposed to not be tested at all, like for example [volume-coupled-diffusion](https://github.com/precice/tutorials/tree/develop/volume-coupled-diffusion), because it can be reasonably assumed that the functionality is already tested via other tutorials. However, this is a choice in the end, which can be changed.

**Comment by BenjaminRodenberg** [comment]
@fsimonis 

### [channel-transport-reaction](https://github.com/precice/tutorials/tree/develop/channel-transport-reaction) chemical-fenics - fluid-fenics 
currently not functional because the two features linear-cell-interpolation and nearest projection mapping do not integrate as they are currently used in the FEniCS adapter. See https://github.com/precice/fenics-adapter/issues/162. This is not a preCICE v3 adapter regression, but a problem with the FEniCS adapter. I would suggest to test [volume-coupled-diffusion](https://github.com/precice/tutorials/tree/develop/volume-coupled-diffusion) instead.

**Comment by BenjaminRodenberg** [comment]
I'm testing the following

| State | Tester | Test | Comment |
| --- | --- | --- | --- |
| | @benjaminrodenberg | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-openfoam - solid-fenics | |
| | @benjaminrodenberg | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-fenics serial + parallel | |
| | @benjaminrodenberg | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-nutils | |
| :x: | @benjaminrodenberg | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) fenics - nutils |added https://github.com/precice/tutorials/commit/be58f19581a65dcc57844a994933886a918be383, problem running https://github.com/precice/tutorials/pull/439 |
| :o: | @benjaminrodenberg | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) fenics - fenics |added https://github.com/precice/tutorials/commit/be58f19581a65dcc57844a994933886a918be383 |
| :x: | @benjaminrodenberg | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) nutils - nutils |added https://github.com/precice/tutorials/commit/be58f19581a65dcc57844a994933886a918be383, problem running `run.sh -d` `./run.sh -n`, https://github.com/precice/tutorials/pull/439 |
|:o: | @benjaminrodenberg | [partitioned-heat-conduction-complex](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction-complex) fenics- fenics | |
| :o: | @benjaminrodenberg | Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy) | added https://github.com/precice/python-bindings/commit/1c25e1324af72bf50deb222fd7e33d2a49e4725c|

with the following SW stack:
* https://github.com/precice/precice/tree/release-v3.0.0 @ b579363cadf11b3bc60e8dcfb6ced09b1923d432
* tutorials @ https://github.com/precice/tutorials/commit/14197c235d4a72d841b04d01446a21a227cf0639
* python-bindings @ https://github.com/precice/python-bindings/releases/tag/v3.0.0.0dev2
* fenics-adapter @ https://github.com/precice/fenics-adapter/commit/6f998598aac79f923235bf1288fd9dd28ae07ef6
* `$ python3 -c "import dolfin; print(dolfin.__version__)"`: `2019.2.0.dev0`
* openfoam-adapter @ https://github.com/precice/openfoam-adapter/commit/99421f77f85e91262f921cf0bbdc3724d0a9cb20
* `$ foamVersion`:`OpenFOAM-v2012`
* `$ python3 -c "import nutils;print(nutils.__version__)"`:`8.4`

**Comment by BenjaminRodenberg** [comment]
@fsimonis :fast_forward: for [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-su2 - solid-fenics ? The SU2 adapter is not ready, but fluid-openfoam - solid-fenics should be a good replacement.

**Comment by MakisH** [comment]
The `precice.integration.Serial` tests are failing on my system (Ubuntu 22.04) with 12a1d168fa:

```
/home/gc/repos/precice/precice/tests/serial/time/explicit/parallel-coupling/DoNonfittingWindows.cpp(65): error: in "Integration/Serial/Time/Explicit/SerialCoupling/DoNonfittingWindows": check nWindows == 2 has failed [1 != 2]
Failure occurred in a following context:
    Test context represents "SolverOne" and runs on rank 0 out of 1.
/home/gc/repos/precice/precice/tests/serial/time/explicit/parallel-coupling/DoNonfittingWindows.cpp(65): error: in "Integration/Serial/Time/Explicit/SerialCoupling/DoNonfittingWindows": check nWindows == 2 has failed [1 != 2]
Failure occurred in a following context:
    Test context represents "SolverTwo" and runs on rank 0 out of 1.
```

**Comment by davidscn** [comment]
I'm testing the following

| State | Tester | Test | Comment |
| --- | --- | --- | --- |
| :o:  | @davidscn |[quickstart](https://github.com/precice/tutorials/tree/master/quickstart) fluid-openfoam - solid-cpp | changed the coupling scheme now to explicit, porting completed and watchpoint looks good 

tbc.

with the following SW stack:
* https://github.com/precice/precice/tree/release-v3.0.0 [9729b7c](https://github.com/precice/precice/commit/9729b7ca028a8e32cd2beb564f642fad5ef37693)
* tutorials develop [09fb08c](https://github.com/precice/tutorials/commit/09fb08c66b2d36219ab7d0c4aa4ec20fceb6cd03)

**Comment by IshaanDesai** [comment]
I tested the following cases (being updated continuously):

| State | Success | Failure | Skipped |
| --- | --- | --- | --- |
|  | @IshaanDesai | [two-scale-heat-conduction](https://github.com/precice/tutorials/tree/develop/two-scale-heat-conduction) macro-nutils - micro-nutils |
| | @IshaanDesai  | [elastic-tube-3d](https://github.com/precice/tutorials/tree/master/elastic-tube-3d) fluid-openfoam - solid-calculix |

Using the software stack:

**Comment by IshaanDesai** [comment]
The test `precice.solverdummy.build.cpp (Failed)` failed on my Ubuntu 22.04 machine with the output:

```bash
Internal cmake changing into directory: /home/desaiin/precice/build/Solverdummies/cpp
======== CMake output     ======
Configuring done
Generating done
Build files have been written to: /home/desaiin/precice/examples/solverdummies/cpp
======== End CMake output ======
Change Dir: /home/desaiin/precice/build/Solverdummies/cpp

Run Clean Command:/usr/bin/gmake -f Makefile clean
gmake: Makefile: No such file or directory
gmake: *** No rule to make target 'Makefile'.  Stop.

Run Build Command(s):/usr/bin/gmake -f Makefile && gmake: Makefile: No such file or directory
gmake: *** No rule to make target 'Makefile'.  Stop.
```

**Comment by MakisH** [comment]
> The `precice.integration.Serial` tests are failing on my system (Ubuntu 22.04) with [12a1d16](https://github.com/precice/precice/commit/12a1d168fa6f20cfda43d46d123b728073f71d76):

All tests work now (f971b4685) on my system.

**Comment by davidscn** [comment]
It would be valuable to add documentation regarding the required polars version (or even prevent too old versions).
`AttributeError: 'DataFrame' object has no attribute 'group_by'. Did you mean: 'groupby'?`

**Comment by uekerman** [comment]
Automatic system tests: https://github.com/precice/tutorials/issues/448

**Comment by fsimonis** [comment]
> The test precice.solverdummy.build.cpp (Failed) failed on my Ubuntu 22.04 machine with the output:

@IshaanDesai This is a problem of CMake itself. `Build files have been written to: /home/desaiin/precice/examples/solverdummies/cpp`

This means that you once manually ran `cmake` inside the `solverdummies/cpp`.
When the tests run, CMake sees that there exists a `CMakeCache.txt` in `solverdummies/cpp` and assumes that it must be the binary directory. Even when explicitly specifying the path as source directory and explicitly specifying another binary directory. Kind of a bug in CMake.

Short term solution `rm solverdummies/cpp/CMakeCache.txt`, long term solution maybe the tests should fail if the CMakeCache.txt is present.

**Comment by MakisH** [comment]
(To be updated and to add an overview table)

## Software stack

- Ubuntu 22.04
- Release branch as of today used from the build directory, with `PKG_CONFIG_PATH` and `LD_LIBRARY_PATH` (needed for the OpenFOAM adapter) set
- OpenFOAM adapter develop as of today (b831b5), OpenFOAM v2312
- CalculiX adapter develop as of today (ffeda1), CalculiX 2.20
- Tutorials develop as of today (60a9794)

## Heat exchanger simplified

{fluid-top-openfoam, solid-calculix, fluid-bottom-openfoam}

### Serial

✅ Serial case works and the results look like the ones in the [tutorial](https://precice.org/tutorials-heat-exchanger-simplified.html).

Screenshot from 2024-01-31 14-01-54

### Parallel

⏳️ Parallel (fluid-top-openfoam:2, solid-calculix:1, fluid-bottom-openfoam:2) is starting.

## Heat exchanger

{fluid-inner-openfoam, solid-calculix, fluid-outer-openfoam}

### Parallel

⏳️ Parallel {fluid-inner-openfoam:2, solid-calculix:1, fluid-outer-openfoam:2} is pending.

## Perpendicular flap

⏳️ {fluid-openfoam:2, solid-calculix:1} pending
⏳️ {fluid-openfoam:1, solid-nutils:1} pending

## Partitioned pipe

⏳️ {fluid1-openfoam-pimplefoam, fluid2-openfoam-pimplefoam} pending
⏳️ {fluid1-openfoam-soniliquidfoam, fluid2-openfoam-sonicliquidfoam} pending

## Channel transport

⏳️ {fluid-openfoam, transport-nutils}

## Fortran module

⏳️ pending

## To be skipped

⏩ code_aster cases (adapter needs to be ported to v3)

**Comment by fsimonis** [comment]
Rebased and cleaned up the release branch.

After the tests from @MakisH go through, we are ready to go.

**Comment by IshaanDesai** [comment]
I ran the perpendicular-flap case with fluid-openfoam and solid-nutils, and I can replicate the behavior that @MakisH is reporting. If I start solid-nutils first, it throws an error and exits. If I start fluid-openfoam first, the tutorial runs fine and the tip displacement plot looks normal.

**Comment by IshaanDesai** [comment]
Regarding the issue of solid-nutils exiting when run first in the perpendicular-flap tutorial, the issue goes away by running `pip3 install --user bottombar==2.1`, which was originally suggested in https://github.com/precice/tutorials/pull/431#pullrequestreview-1812711006 by @uekerman.

**Comment by MakisH** [comment]
> Regarding the issue of solid-nutils exiting when run first in the perpendicular-flap tutorial, the issue goes away by running `pip3 install --user bottombar==2.1`, which was originally suggested in [precice/tutorials#431 (review)](https://github.com/precice/tutorials/pull/431#pullrequestreview-1812711006) by @uekerman.

✅ I can confirm that the case works with Nutils 8.5 (bottombar 2.1.0). Let's continue the discussion in the issue: https://github.com/precice/tutorials/issues/457

I tested several different cases, I think we are good to go.

Unless we really need to move the fortran module to now, while it is supposed to be in the post-release section.

**Comment by fsimonis** [comment]
The release PR is now ready for review!

**Review by BenjaminRodenberg — DISMISSED** [review]

**Review by IshaanDesai — DISMISSED** [review]

**Review by MakisH — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
         working-directory: build
         run: |
           cmake --version
-          cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCPACK_GENERATOR="DEB" -DPRECICE_PETScMapping=ON -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON ..
+          cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCPACK_GENERATOR="DEB" -DPRECICE_FEATURE_PETSC_MAPPING=ON -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON ..
```
Comment (davidscn) [inline .github/workflows/release.yml:48]: I'm not sure about benefits, but we could consider enabling `CMAKE_INTERPROCEDRUAL_OPTIMIZATION` for the package generation (we also highlight this in our CMake variables).

**Review by davidscn — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (fsimonis) [reply .github/workflows/release.yml:48]: True.

When we drop Ubuntu 20.04, our new cmake baseline is 3.22 and then we can use a preset for this, which were introduced with version 3.19.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,151 @@
...
+- Added CMake presets for simpler configuration. (https://github.com/precice/precice/pull/1452)
+- Added Wendland's compactly supported C2 and C4 RBF functions. (https://github.com/precice/precice/pull/1378)
+- Added `<mapping:rbf>` alias defaulting to the `rbf-pum-direct`. (https://github.com/precice/precice/pull/1616)
+- Added `precicef_get_version_information_` to Fortran bindings. (https://github.com/precice/precice/pull/1487)
+- Added a `scaled-consistent-volume` mapping constraint in order to scale by volumetric primitives. (https://github.com/precice/precice/pull/1387)
+- Added a mapping configuration subtag `executor` to specify where and how RBF data mappings should be computed: `cpu` (default), `cuda`, `hip`, or `openmp`. (https://github.com/precice/precice/pull/1630)
+- Added a new mixed convergence measure `absolute-or-relative-convergence-measure`. (https://github.com/precice/precice/pull/1862)
+- Added a warning for empty IQN matrix, but keep the simulation going. This allows to start from a zero initial state that only later changes (e.g., an opening valve in a flow simulation) (https://github.com/precice/precice/pull/1895)
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply CHANGELOG.md:15]: This looks a bit odd indeed.

If we remove the entries from here, then the wording doesn't make sense for the hotfix. As it mentions backporting changes. 

Unsure what the right way forward is here. More opinions welcome.

[the remark this answers is not in the document]
Comment (davidscn) [reply CHANGELOG.md:15]: >  As it mentions backporting changes.

Hm I see. In some sense we didn't do a backporting, as the new changes were not released yet, but looking at the release message again, I guess it makes sense to keep the entries. In the release notes, we explicitly refer to the future release.

[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply CHANGELOG.md:15]: The "backporting" is only the way how it internally developed. But I think this kind of history is not really relevant for the users. What they see: We did a bugfix via 2.5.1. Do we even need to mention it again in 3.0.0?

**Review by carme-hp — DISMISSED** [review]

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fsimonis) [reply CHANGELOG.md:15]: Given that many people will only use/look at 2.5.1 if they need to stay on 2.x, I would leave the entries in the changelog of 3.0.0.

The formulation of "back-porting" emphasizes that these fixes required extra work to be applied to 2.5.0. And that the 2 series isn't in active development any more.

Let's leave it this way

**Review by Fujikawas — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (davidscn) [reply .github/workflows/release.yml:48]: Still a valuable switch. Maybe also for the request debian packages workflow..

**Review by IshaanDesai — APPROVED** [review]

**Review by BenjaminRodenberg — APPROVED** [review]

**Pull request merged**